metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableUtil.java

```diff
@@ -0,0 +1,196 @@
+ */
+
+package com.netflix.metacat.connector.hive.iceberg;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.netflix.iceberg.BaseMetastoreTableOperations;
+import com.netflix.iceberg.BaseMetastoreTables;
+import com.netflix.iceberg.PartitionSpec;
+import com.netflix.iceberg.ScanSummary;
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.Table;
+import com.netflix.iceberg.TableMetadata;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.exception.MetacatNotSupportedException;
+import com.netflix.metacat.common.server.connectors.ConnectorContext;
+import com.netflix.metacat.common.server.connectors.model.PartitionListRequest;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.Map;
+
+/**
+ * Iceberg Table Util.
+ *
+ * @author zhenl
+ * @since 1.2.0
+ */
+public class IcebergTableUtil {
+    private final Configuration conf;
+    private final ConnectorContext connectorContext;
+
+    /**
+     * Constructor.
+     *
+     * @param connectorContext connector context
+     */
+    public IcebergTableUtil(final ConnectorContext connectorContext) {
+        this.conf = new Configuration();
+        this.connectorContext = connectorContext;
+        connectorContext.getConfiguration().keySet()
+            .forEach(key -> conf.set(key, connectorContext.getConfiguration().get(key)));
+    }
+
+    /**
+     * get Partition Map.
+     *
+     * @param icebergTable      iceberg Table
+     * @param partitionsRequest partitionsRequest
+     * @return partition map
+     */
+    public Map<String, ScanSummary.PartitionMetrics> getIcebergTablePartitionMap(
+        final Table icebergTable,
+        final PartitionListRequest partitionsRequest) {
+        return (partitionsRequest.getPageable() != null)
+            ? ScanSummary.of(icebergTable.newScan()).build() //whole table scan with paging
+            :
+            ScanSummary.of(icebergTable.newScan())  //the top x records
+                .limit(connectorContext.getConfig().getIcebergTableSummaryFetchSize())
+                .build();
+    }
+
+    /**
+     * get iceberg table.
+     *
+     * @param tableName             table name
+     * @param tableMetadataLocation table metadata location
+     * @return iceberg table
+     */
+    public Table getIcebergTable(final QualifiedName tableName, final String tableMetadataLocation) {
+        return new IcebergMetastoreTables(tableMetadataLocation).load(tableName.toString());
+    }
+
+    /**
+     * get data metadata from partition metrics.
+     *
+     * @param metrics metrics.
+     * @return object node of the metrics
+     */
+    public ObjectNode getDataMetadataFromIcebergMetrics(
+        final ScanSummary.PartitionMetrics metrics) {
+        final ObjectNode root = JsonNodeFactory.instance.objectNode();
+        root.set(DataMetricConstants.DATA_METADATA_METRIC_NAME, getMetricValueNode(metrics));
```

Comment:
This is fine for now. We should have used the dse-metric library to set the values, instead of manipulating the JSON.

Reply:
is dse-metric library open sourced? we may want to stay like this if not.